songhyun/app.js

```diff
@@ -49,6 +50,50 @@ app.post("/join", async (req, res) => {
   res.status(201).json({ message: "user created" });
 });
 
+app.post("/login", async (req, res) => {
+  const { name, password } = req.body;
+
+  const hashedPassword = await myDataSource.query(
+    `SELECT 
+      users.password,
+      users.id
+    FROM users
+    WHERE users.name="${name}";
+    `
+  );
+  const dbPassword = hashedPassword[0].password;
+  const userId = hashedPassword[0].id;
+  const check = await bcrypt.compare(password, dbPassword);
+  if (!check) {
+    res.json({ message: "Invalid User" });
+  }
+
+  const send = { userId: userId };
+  const jwtToken = jwt.sign(send, process.env.secretKey);
+  res.json({ accessToken: jwtToken });
+});
+
+app.post("/post", async (req, res) => {
+  const { title, content } = req.body;
+  const token = req.headers.token;
+  try {
+    const decoded = jwt.verify(token, process.env.secretKey);
+    if (decoded) {
+      await myDataSource.query(
+        `INSERT INTO posts(
+          title,
+          content,
+          user_id
+        ) VALUES (?, ?, ?);`,
+        [title, content, decoded.userId]
+      );
+      res.status(201).json({ message: "postCreated" });
+    }
+  } catch (err) {
+    res.status(401).json({ message: "Invalid Access Token" });
+  }
```

Comment:
```suggestion
  try {
    const decoded = jwt.verify(token, process.env.secretKey);

    await myDataSource.query(
      `INSERT INTO posts(
        title,
        content,
        user_id
        ) VALUES (?, ?, ?);`,
        [title, content, decoded.userId]
      );
      
      res.status(201).json({ message: "postCreated" });
    }
  } catch (err) {
    res.status(401).json({ message: "Invalid Access Token" });
  }
```
애초에 decode가 안되는 경우엔 다른 에러가 발생하지 않나요?

만약 다른 에러가 throw 되는 것이 아니라서 이렇게 작성하신거라면

early return 패턴을 적용해 보세요.

```JavaScript
const decoded = jwt.verify(...)

// 문제가 생길만한 부분을 따로 로직으로 작성하여 처리 합니다.
// 이후에 작성되는 케이스는 모두 개발자가 의도한 성공 케이스입니다.
if (!decoded) {
  // 여기서 throw된 error를 try catch로 잡아주면 됩니다.
   throw new Error("..")
}

await AppDataSource.query(`...`)

return res.status(...)
```